src/ace.F90

```diff
@@ -409,6 +414,69 @@ subroutine read_ace_table(i_table, i_listing)
   end subroutine read_ace_table
 
 !===============================================================================
+! READ_MULTIPOLE_DATA checks for the existence of a multipole library in the
+! directory and loads it using multipole_read
+!===============================================================================
+
+  subroutine read_multipole_data(i_table)
+
+    integer, intent(in) :: i_table   ! index in nuclides/sab_tables
+
+    logical       :: file_exists   ! does multipole library exist?
+    character(7)  :: readable      ! is multipole library readable?
+    character(6)  :: zaid_string   ! String of the ZAID
+    character(9)  :: filename      ! path to multipole cross section library
+    type(Nuclide),   pointer :: nuc => null()
+
+    ! For the time being, and I know this is a bit hacky, we just assume
+    ! that the file will be zaid.h5.
+    nuc => nuclides(i_table)
+
+    write(zaid_string,'(I6.6)') nuc % zaid
+    filename = zaid_string // ".h5"
+
+    ! Check if Multipole library exists and is readable
+    inquire(FILE=filename, EXIST=file_exists, READ=readable)
+    if (.not. file_exists) then
+      nuc % mp_present = .FALSE.
+      return
+    elseif (readable(1:3) == 'NO') then
+      call fatal_error("Multipole library '" // trim(filename) // "' is not readable! &
+           &Change file permissions with chmod command.")
+    end if
+
+    ! display message
+    call write_message("Loading Multipole XS table: " // filename, 6)
+
+    allocate(nuc % multipole)
+
+    ! Call the read routine
+    call multipole_read(filename, nuc % multipole, i_table)
+    nuc % mp_present = .TRUE.
+
+    ! Update the maximum number of poles, l indices, and polynomial order
+    if(nuc % multipole % max_w > max_poles) then
+      max_poles = nuc % multipole % max_w
+    end if
+
+    if(nuc % multipole % num_l > max_L) then
+      max_L = nuc % multipole % num_l
+    end if
+
+    if(nuc % multipole % fit_order + 1 > max_poly) then
+      max_poly = nuc % multipole % fit_order + 1
+    end if
+
+    ! Recreate nu-fission tables
+    if(nuc % fissionable) then
```

Comment:
All of the `if`s above should have a space after the `if`, e.g. `if (nuc%fissionable) then`